### DISCUSSION TITLE
Support for wss

### DISCUSSION THREAD

**Discussion by Ancient-Dragon** [graphql-java-kickstart/graphql-java-servlet#361] [Q&A]
Hi there,

I'm looking to use subscriptions in my application, this works locally, but as soon as I put this up into different environments which require things like https etc. I am unable to get subscriptions work. If I use the ws I get an error from chrome etc saying that https to unsecure web sockets is depreciated (understandable). However, when I try to use wss I get a 404? What have other people done to handle this?

Thanks.

**Comment by jamesmfriedman** [comment]
My team is only running in wss mode and have never gotten this to work. We've wasted quite a bit of time on this at this point. Am I to understand that wss just flat out doesn't work?

**Comment by oliemansm** [comment]
I'm not familiar with your exact configuration to understand what you're running into. We never expose the Java application directly but instead use a setup where we have a webserver like Nginx in front of our Java app. Nginx is responsible for the SSL connection including the wss part. All traffic is forwarded to the non-SSL counterpart of the Java app. This works fine in our case.

**Reply by jamesmfriedman** [reply]
Thanks @oliemansm thats basically my guidance for a last attempt at getting this working. I believe we do SSL termination higher up in the stack in production, but for local dev the setup might be different. Will give it a shot, thank you for responding!
